src/app/pages/class/class.component.spec.ts

```diff
@@ -129,12 +130,11 @@ describe('class component', () => {
     expect(fixture.componentInstance.dataSource.data.length).toEqual(23);
   });
   it('should load zero classes in case of promise reject', async () => {
-    (classServiceMock.getAllClasses as jest.Mock).mockImplementationOnce(() => {
-      return Promise.resolve(Promise.reject());
-    });
+    const getAllClassesMock = classServiceMock.getAllClasses as jest.Mock;
+    getAllClassesMock.mockReset();
+    getAllClassesMock.mockRejectedValueOnce(0);
+
     const fixture = TestBed.createComponent(ClassComponent);
-    fixture.detectChanges();
```

Comment:
make sure we realy don't need this.

Reply:
Even though this is not strictly needed, indeed it would be best practice to detect changes and wait for the rendering to be done, so I have put that code back in.